Test/Invoke-WhiskeyNpmPrune.Tests.ps1

```diff
@@ -0,0 +1,259 @@
+}
+
+function GivenFailingNpm
+{
+    param (
+        $ExitCode
+    )
+
+    Mock -CommandName 'Initialize-NodeProject'
+    Mock -CommandName 'Invoke-Command' -ModuleName 'Whiskey' -ParameterFilter { $ScriptBlock -match 'prune' } -MockWith { & cmd /c exit 1 }
+}
+
+function GivenMissingPackageJson
+{
+    Mock -CommandName 'Initialize-NodeProject'
+}
+
+function GivenWorkingDirectory
+{
+    param(
+        $Directory
+    )
+    $script:givenWorkingDirectory = $Directory
+    $script:workingDirectory = Join-Path -Path $workingDirectory -ChildPath $Directory
+
+    New-Item -Path $workingDirectory -ItemType 'Directory' -Force | Out-Null
+}
+
+function WhenRunningTask
+{
+    [CmdletBinding()]
+    param()
+
+    $taskContext = New-WhiskeyTestContext -ForBuildServer -ForBuildRoot $TestDrive.FullName
+
+    $taskParameter = @{ 'NpmRegistryUri' = $script:npmRegistryUri }
+
+    if ($givenWorkingDirectory)
+    {
+        $taskParameter['WorkingDirectory'] = $givenWorkingDirectory
+    }
+
+    if ($initialize)
+    {
+        $taskContext.RunMode = 'Initialize'
+    }
+
+    Push-Location $script:workingDirectory
+
+    try
+    {
+        Initialize-NodeProject
+
+        Invoke-WhiskeyTask -TaskContext $taskContext -Parameter $taskParameter -Name 'NpmPrune'
+    }
+    catch
+    {
+        $script:failed = $true
+        Write-Error -ErrorRecord $_
+    }
+    finally
+    {
+        Pop-Location
+    }
+}
+
+function ThenNpmPruneNotCalled
+{
+    It 'should not run ''npm prune''' {
+        Assert-MockCalled -CommandName 'Invoke-Command' -ModuleName 'Whiskey' -ParameterFilter { $ScriptBlock -match 'prune' } -Times 0
+    }
+}
+
+function ThenPackage
+{
+    param(
+        [Parameter(Position=0)]
+        [string]
+        $PackageName,
+        
+        [Parameter(Mandatory=$true,ParameterSetName='Exists')]
+        [switch]
+        $Exists,
+
+        [Parameter(Mandatory=$true,ParameterSetName='DoesNotExist')]
+        [switch]
+        $DoesNotExist
+    )
+
+    $packagePath = Join-Path -Path $script:workingDirectory -ChildPath ('node_modules\{0}' -f $PackageName)
+
+    If ($Exists)
+    {
+        It ('should not prune package ''{0}''' -f $PackageName) {
+            $packagePath | Should -Exist
+        }
+    }
+    else
+    {
+        It ('should prune package ''{0}''' -f $PackageName) {
+            $packagePath | Should -Not -Exist
+        }
+    }
+}
+
+function ThenTaskFailedWithMessage
+{
+    param(
+        $Message
+    )
+
+    It 'task should fail' {
+        $failed | Should -Be $true
+    }
+
+    It ('error message should match [{0}]' -f $Message) {
+        $Global:Error[0] | Should -Match $Message
+    }
+}
+
+function ThenTaskSucceeded
+{
+    It 'should not write any errors' {
+        $Global:Error | Should -BeNullOrEmpty
+    }
+
+    It 'should not fail' {
+        $failed | Should -Be $false
+    }
+}
+
+Describe 'NpmPrune.when running in Initialize mode' {
```

Comment:
Add a `Then` assertion that checks if the task really initialized, i.e. if Node.js was installed. I would probably mock out the call to 'Install-WhiskeyNodeJs`.

Reply:
Good catch. Fixed.